File: algorithms/ar-mexp/C++/mexp.cpp
#include<bits/stdc++.h>

using namespace std;

int modular_exponentiation(int base,int power,int mod)
{
    int result=1;
    while(power>0)
    {
	if(power&1)
	{
	    result=(result*base)%mod;
	}
	base=(base*base)%mod;
	power>>=1;
    }
    return result;	
}

int main()
{
	cout<<modular_exponentiation(2,3,5)<<endl;
}

File: algorithms/ar-mexp/PHP/mexp.php
<?php

function mod_ex( int $base, int $power, int $mod ) {
    $iResult = 1;
    while( $power > 0 ) {
        if( $power & 1 ) {
            $iResult = ( $iResult * $base ) % $mod;
        }
        $base = ( $base * $base ) % $mod;
        $power >>= 1;
    }
    return $iResult;
}

var_dump( mod_ex( 2,6,5 ) );